### PR TITLE
MOB-1150 - Animated PFP Support

### PR DESCRIPTION
**Pull request by Oleg-Pecheneg**

### CODE REVIEW

File: unstoppable-ios-app/domains-manager-ios/Modules/DomainProfile/Cells/DomainProfileTopInfoCell/DomainProfileTopInfoCell.swift [hunk 1]
```diff
@@ -135,7 +135,8 @@ extension DomainProfileTopInfoCell {
                 case .image(let image, _):
                     imageView.image = image
                 case .imageURL(let url, _):
-                    let image = await appContext.imageLoadingService.loadImage(from: .url(url),
+                    let image = await appContext.imageLoadingService.loadImage(from: .url(url,
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Modules/DomainProfile/Cells/DomainProfileTopInfoCell/DomainProfileTopInfoCell.swift:138]: 5 or more calls with a new parameter and the same value...

wouldn't be nicer to make `Constants.downloadedImageMaxSize` a default value for `maxSize`?

File: unstoppable-ios-app/domains-manager-ios/Services/GIFAnimationsService.swift [hunk 2]
```diff
@@ -59,6 +59,22 @@ extension GIFAnimationsService {
             await stateHolder.removeGIF(gif)
         }
     }
+    
+    func gifImageWithData(_ data: Data,
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Services/GIFAnimationsService.swift:63]: **create**GifImage(from: Data) ?

File: unstoppable-ios-app/domains-manager-ios/Services/GIFAnimationsService.swift [hunk 3]
```diff
@@ -70,12 +86,19 @@ extension GIFAnimationsService {
...
+            case .happyEnd: return .maskWhite
+            }
+        }
     }
 }
 
 // MARK: - GIF animations. Use GIFAnimationsService to work with GIF animations
 private extension GIFAnimationsService {
-    func gifImageWithURL(_ gifUrl:String) async -> UIImage? {
+    func gifImageWithURL(_ gifUrl: String,
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Services/GIFAnimationsService.swift:100]: fetch?

File: unstoppable-ios-app/domains-manager-ios/Services/GIFAnimationsService.swift [hunk 4]
```diff
@@ -85,10 +108,11 @@ private extension GIFAnimationsService {
             return nil
         }
         
-        return await gifImageWithData(imageData)
+        return await gifImageWithData(imageData, maskingType: maskingType)
     }
     
-    func gifImageWithName(_ name: String) async -> UIImage? {
+    func gifImageWithName(_ name: String,
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Services/GIFAnimationsService.swift:114]: fetch?

File: unstoppable-ios-app/domains-manager-ios/Services/GIFAnimationsService.swift [hunk 5]
```diff
@@ -206,6 +223,18 @@ private extension GIFAnimationsService {
         return (images, delays)
     }
     
+    func maskingImage(_ image: CGImage, withMaskingType maskingType: GIFMaskingType?) -> CGImage? {
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Services/GIFAnimationsService.swift:226]: get? create? build?

File: unstoppable-ios-app/domains-manager-ios/Services/ImageLoadingService.swift [hunk 6]
```diff
@@ -144,27 +143,34 @@ extension ImageLoadingService: ImageLoadingServiceProtocol {
 
 // MARK: - Private methods
 fileprivate extension ImageLoadingService {
-    func imageFor(source: ImageSource, shouldCache: Bool = true, downsampleDescription: DownsampleDescription?) async -> UIImage? {
+    func imageFor(source: ImageSource, downsampleDescription: DownsampleDescription?) async -> UIImage? {
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Services/ImageLoadingService.swift:146]: get? fetch?

**Review by rommex — APPROVED** [review]
Please consider renaming methods unless this is forced by an external protocol

File: unstoppable-ios-app/domains-manager-ios/Modules/DomainProfile/Cells/DomainProfileTopInfoCell/DomainProfileTopInfoCell.swift (hunk 1 above)
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Modules/DomainProfile/Cells/DomainProfileTopInfoCell/DomainProfileTopInfoCell.swift:138]: @rommex They're similar but not the same, `downloadedImageMaxSize` and `downloadedIconMaxSize`

**Pull request merged**